File: app/src/main/java/com/example/youtubeapiintegration/Retrofit/RetrofitInstance.java
package com.example.youtubeapiintegration.Retrofit;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitInstance {

    private static Retrofit retrofit = null;

    private static String BASE_URL = "https://www.googleapis.com/youtube/v3/";

    public static Retrofit getRetrofit() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .addConverterFactory(GsonConverterFactory.create())
                    .baseUrl(BASE_URL)
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/com/example/youtubeapiintegration/Retrofit/RetrofitInstance.kt
package com.example.youtubeapiintegration.Retrofit

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitInstance {

    private var retrofit: Retrofit? = null

    private val BASE_URL = "https://www.googleapis.com/youtube/v3/"

    fun getRetrofit(): Retrofit? {
        if (retrofit == null) {
            retrofit = Retrofit.Builder()
                    .addConverterFactory(GsonConverterFactory.create())
                    .baseUrl(BASE_URL)
                    .build()
        }
        return retrofit
    }
}
